### PR TITLE
CC-538/DRYD-831: Filtering which MIME type options are shown.

### PR DESCRIPTION
**Pull request by cesarvh**

### CODE REVIEW

**Comment by cesarvh** [comment]
Currently working on coverage fixes for this PR.

File: src/components/invocable/InvocationModal.jsx [hunk 1]
```diff
@@ -225,28 +230,44 @@ export default class InvocationModal extends Component {
...
     } = this.state;
 
-    // TODO: Use reports/mimetypes endpoint to get options, and [recordType]/mimetypes to check if
-    // a format picker should be shown instead of hardcoding (batch/mimetypes will return 404, so
-    // format picker should not be shown).
-
     if (recordType === 'report') {
+      let mimeList = [];
+
+      if (getMimeTypes) {
```
Comment (ray-lee) [inline src/components/invocable/InvocationModal.jsx]: Don't need this. The report data is already in the `data` prop. Also React render methods should not have side effects, so if you needed to make a request, this wouldn't be the place to do it.

**Comment by cesarvh** [comment]
Forced push to delete a few commits I added on accident from another issue.

**Comment by cesarvh** [comment]
Hi Ray,

A couple months back, @remillet  made a change to the report schema [here](https://github.com/collectionspace/services/pull/232). We created a report on botgarden-dev with these new fields. I made it backwards compatible with reports that don't have these fields, too. 

If the files are present, the resource record looks loke
```xml
<?xml version="1.0"?>
<ns2:reports_common xmlns:ns2="http://collectionspace.org/services/report" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <notes>
Individuals that died in a specified garden location
</notes>
  <filename>ucbgDiedInLocationExcel.jasper</filename>
  <forDocTypes>
    <forDocType>Locationitem</forDocType>
  </forDocTypes>
  <supportsGroup>false</supportsGroup>
  <name>Deads in Bed (Excel)</name>
  <supportsOutputMIMEList>
    <outputMIME>
application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
</outputMIME>
  </supportsOutputMIMEList>
  <supportsSingleDoc>true</supportsSingleDoc>
  <outputMIME>
application/vnd.openxmlformats-officedocument.spreadsheetml.sheet
</outputMIME>
  <supportsDocList>false</supportsDocList>
  <supportsNoContext>false</supportsNoContext>
  <forRoles/>
  <resourceActionGroupList/>
</ns2:reports_common>
```
And the UI presents only the XML MIME type:
image

If we go ahead and remove it from the resource record, we get an unfiltered list:
image

**Comment by ray-lee** [comment]
@cesarvh  @remillet It sounds like you got it working on the UCB botgarden branch. Have you actually tried it on a build from collectionspace/master? I don't think it works.

**Comment by cesarvh** [comment]
@ray-lee Since we don't have tomcat8 yet, we can't deploy the latest services layer code. Is the `https://core.dev.collectionspace.org/` services layer code the most up to date?

**Comment by ray-lee** [comment]
@cesarvh Yes, core.dev has the latest. If you're developing contributions for master, I expect that you have tested it against a build from master before you make a pull request. If it's UI code, you can just use  *.dev.collectionspace.org as the back end. If you're developing services layer code, you have to have a local installation built from master. Use your own VM, if you have to.

**Comment by ray-lee** [comment]
@Cesar, the problem was that app layer config was missing for the supportsOutputMIMEList field. You have this in the UCB branch, but you/Richard did not put it into master. That's why it's so important to actually test that your pull request works on master. I've fixed this now.

**Pull request merged**